hazelcast/src/main/java/com/hazelcast/map/impl/MapSplitBrainHandlerService.java

```diff
@@ -58,14 +58,14 @@ protected Runnable newMergeRunnable(Collection<RecordStore> mergingStores) {
     protected void onStoreCollection(RecordStore recordStore) {
         assertRunningOnPartitionThread();
 
-        ((DefaultRecordStore) recordStore).clearIndexedData(false);
+        ((DefaultRecordStore) recordStore).clearOtherResourcesThanStorage(true);
     }
 
     @Override
     protected void destroyStore(RecordStore store) {
         assertRunningOnPartitionThread();
 
-        store.destroyInternals();
+        ((DefaultRecordStore) store).clearAndDestroyStorage(false, true);
```

Comment:
Here we also have a difference in behaviour. Previously:
```
        clearMapStore();
        clearStorage(false);
        storage.destroy(false);
        mutationObserver.onDestroy(true);
```

Now:
```
        storage.clear(false);
        mutationObserver.onClear();
        storage.destroy(false);
        mutationObserver.onDestroy(true);
```
The difference is that we clear the mutation observers. This might be ok if the observer data has already been merged or if the observer has no longer a reference to the data. Previously the observer data was not cleared or destroyed. Also, we don't clear the map store anymore.

Reply:
Double-checked this comment, seems ok. Thanks for the explanation.